### DISCUSSION TITLE
Can't get User System Command Function to work

### DISCUSSION THREAD

**Discussion by Ethanil** [SilentVoid13/Templater#421] [Help]
Hey!

I am trying to use [gcalcli](https://github.com/insanum/gcalcli) to create a calendar entry.
I can create the agenda without a problem in the Terminal but i can't get Templater to insert the output into my new file.
Here are my settings:
Name of the Function: gcalcli
User Function: `gcalcli  --calendar uni agenda`
-> I also tried `/usr/bin/gcalcli`, both work in the Terminal...
My Template is just: `<% tp.user.gcalcli() %>`

I get the following Error Message:
```
Templater Error: Error with User Template gcalcli 
 Error: Command failed: gcalcli  --calendar uni agenda
/bin/sh: line 1: gcalcli: command not found
```
or 
```
Templater Error: Error with User Template gcalcli 
 Error: Command failed: /usr/bin/gcalcli  --calendar uni agenda
/bin/sh: line 1: /usr/bin/gcalcli: No such file or directory
```
When using the full path to gcalcli

using stuff like `echo "test"` does work, i have no idea what i should try next

**Comment by AB1908** [comment]
If this is still an issue, can you try using shabegom's example snippets?

**Reply by 1nsp1r3rnzt** [reply]
I am having same error. Were you able to fix it?

**Reply by AB1908** [reply]
How strange. Can you check if `/bin/sh /usr/bin/gcalcli` actually works?

**Reply by monkeyxite** [reply]
Same issue happens here for khal, task warrior integration. Have to reply on full path feed into User definition functions, and apparently even though defines the shell (with zshrc and relatvant path exported) but still could not find the commands under zsh's paths.
